File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> hashMap = new HashMap<>(); 
        for(int i = 0; i < nums.length; i++){
            int targetSum = target - nums[i]; 
            if(hashMap.containsKey(targetSum)) return new int[]{hashMap.get(targetSum), i}; 
            hashMap.put(nums[i],i); 
        }
        return new int[]{}; 
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> answer; 
        map<int,int> hashMap; 
        for(int i = 0; i < nums.size(); i++){
            int find = target - nums[i]; 
            if(hashMap.count(find)){
                return {hashMap[find], i}; 
            } 
            hashMap[nums[i]] = i; 
        }
        return {}; 
    }
};